### ISSUE TITLE
Problems installing BluePyOpt

### ISSUE THREAD

**Issue by SvennoNito** [BlueBrain/BluePyOpt#227]
Hey,
I'm very happy you've build this toolbox but since I have set up a python 2.7 kernel on my Jupyter Notebook I can't install BluePyOpt anymore. I've explained my problem in more detail [here](https://stackoverflow.com/questions/46424265/python-installing-a-module-successfuly-but-it-is-not-available-via-import). I hope you guys can help!

**Comment by mgeplf** [comment]
It looks like it's an efel (one of the BluePyOpt dependencies) that's not compiling properly on windows.  The part of the error message that might help is:
```
error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools
```

Unfortunately, I don't have a windows machine to test such things on, so I'm not sure that it will work, but that's what I'd try first.

**Comment by wvangeit** [comment]
@SvennoNito do you have Microsoft visual c++ compilers installed on your system ?

**Comment by SvennoNito** [comment]
Indeed, that seems to solve it! At least in my python 3 kernel. Which is nice! But in my python 2.7 kernel I still get the same output when installing it. Weird.

**Comment by SvennoNito** [comment]
By the way, I only need python 2.7 for NEURON I think right? Is it worth rewriting all the print " " commands as print(" ") or are there other things that interfere with python 3?

**Comment by wvangeit** [comment]
Actually, you don't need python 2.7 for NEURON either. BluePyOpt, eFEL and NEURON are all compatible with python 3.

**Comment by SvennoNito** [comment]
Oh, but when I run
`nrn = ephys.simulators.NrnSimulator()`
I get this error
```
  File "C:\Users\Alex\Anaconda3\lib\site-packages\neuron\__init__.py", line 90
    print "Found NEURON at", neuronhome
                          ^
SyntaxError: Missing parentheses in call to 'print'
```

**Comment by wvangeit** [comment]
Mmm, it looks like the wrong version of neuron is installed on your computer. This seems to be the python 2 version.

**Comment by SvennoNito** [comment]
Wait, do I need to run the .exe file or do I ned to pip install NEURON in python?
If I don't pip install it in python it says no module named NEURON available.

**Comment by SvennoNito** [comment]
and if I pip install NEURON then I get the print error :/

**Comment by wvangeit** [comment]
No, pip installing neuron won't work. I think it's best to ask the Neuron developers how to install it on Windows in python3.
Another option would be that you try to fix the compiler issue in the python 2 environment.

**Comment by SvennoNito** [comment]
Oki I hope they'lkl respond as quick as you guys :-)

**Comment by SvennoNito** [comment]
So I managed to change the error message :D Don't know whether it is good though. So [here](http://neuron.yale.edu/neuron/static/docs/neuronpython/firststeps.html#step-1-import-the-neuron-module-into-python) it says to import NEURON as

`from neuron import h, gui`

and indeed when I run 

`responses = twostep_protocol.run(cell_model=simple_cell, param_values=default_params, sim=nrn)  # run the protocoll on a cell`

now it nolonger says "no module named NEURON" but now it says 

`Exception: Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\bluepyopt\ephys\protocols.py", line 130, in _run_func
    cell_model.instantiate(sim=sim)
  File "C:\Anaconda3\lib\site-packages\bluepyopt\ephys\models.py", line 222, in instantiate
    self.morphology.instantiate(sim=sim, icell=self.icell)
  File "C:\Anaconda3\lib\site-packages\bluepyopt\ephys\morphologies.py", line 91, in instantiate
    self.morphology_path)
OSError: Morphology not found at 'simple.swc'`

The NEURON guys haven't even accepted my account for their forum yet so I was hoping that's maybe something again that you guys are familiar with. As always, any idea is highly appreciated :-)

**Comment by wvangeit** [comment]
Is the 'simple.swc' file located in the directory you are running this from ?

**Comment by SvennoNito** [comment]
How can I check this?

**Comment by wvangeit** [comment]
Are you loading an ipython notebook ? Does the directory that contains the notebook also have the file simple.swc ?
From where are you getting the files ?

**Comment by SvennoNito** [comment]
Yes I'm working from Jupyter Notebook.
I'm not exactly sure what you're asking for (sorry for that, I'm still very new to python). I'll give you some info hoping that's what you're asking for. To be honest I don't know how to check where python stores simple.swc (?). I'm using your example script you're providing on github by the way. I think I'm loading it via 

`morph = ephys.morphologies.NrnFileMorphology('simple.swc')     # A morphology can be loaded from a file (SWC or ASC).`

and this works. Anyway, here some path infos:

`!jupyter kernelspec list`
Available kernels:
  python3    C:\Anaconda3\share\jupyter\kernels\python3

`import sys
sys.executable`
'C:\\Anaconda3\\python.exe'

`sys.path`
['',
 'c:\\nrn\\lib\\python',
 'C:\\Anaconda3\\python36.zip',
 'C:\\Anaconda3\\DLLs',
 'C:\\Anaconda3\\lib',
 'C:\\Anaconda3',
 'C:\\Anaconda3\\lib\\site-packages',
 'C:\\Anaconda3\\lib\\site-packages\\Babel-2.5.0-py3.6.egg',
 'C:\\Anaconda3\\lib\\site-packages\\win32',
 'C:\\Anaconda3\\lib\\site-packages\\win32\\lib',
 'C:\\Anaconda3\\lib\\site-packages\\Pythonwin',
 'C:\\Anaconda3\\lib\\site-packages\\IPython\\extensions',
 'C:\\Users\\SvenLeach\\.ipython']

**Comment by wvangeit** [comment]
I just realised we forgot to answer your last questions. Do you still have this issue ?

**Comment by SvennoNito** [comment]
I did, which is why I swtiched to a Linux cluster. I couldn't reach anyone from Neuron. The cluster is administered by my University, they took care of installing NEURON on the cluster, now it works there. On my own Windows 10 PC, however, I couldn't figure out how to install NEURON. It's not an issue for me at the moment though, as I had the opportunity to work on the cluster.

In the meantime I tried to implement all the EA functions by myself, which turned out to be a mess! I'm really grateful that you guys developed a toolbox for this. I hope I'll understand its usage quickly :-)
